simple-local-avatars.dev.js

```diff
@@ -1,65 +1,164 @@
-			button: { text: i10n_SimpleLocalAvatars.insertIntoPost },
-			library : { type : 'image'},
-			multiple: false
+		/* We need to setup a Crop control that contains a few parameters
+		and a method to indicate if the CropController can skip cropping the image.
+		In this example I am just creating a control on the fly with the expected properties.
+		However, the controls used by WordPress Admin are api.CroppedImageControl and api.SiteIconControl
+		*/
+
+		const cropControl = {
+			id: 'control-id',
+			params: {
+				flex_width: false, // set to true if the width of the cropped image can be different to the width defined here
+				flex_height: false, // set to true if the height of the cropped image can be different to the height defined here
+				width: 200, // set the desired width of the destination image here
+				height: 200, // set the desired height of the destination image here
+			},
+		};
+
+		cropControl.mustBeCropped = function (flexW, flexH, dstW, dstH, imgW, imgH) {
+			// If the width and height are both flexible
+			// then the user does not need to crop the image.
+
+			if ( true === flexW && true === flexH ) {
+				return false;
+			}
+
+			// If the width is flexible and the cropped image height matches the current image height,
+			// then the user does not need to crop the image.
+			if ( true === flexW && dstH === imgH ) {
+				return false;
+			}
+
+			// If the height is flexible and the cropped image width matches the current image width,
+			// then the user does not need to crop the image.
+			if ( true === flexH && dstW === imgW ) {
+				return false;
+			}
+
+			// If the cropped image width matches the current image width,
+			// and the cropped image height matches the current image height
+			// then the user does not need to crop the image.
+			if ( dstW === imgW && dstH === imgH ) {
+				return false;
+			}
+
+			// If the destination width is equal to or greater than the cropped image width
+			// then the user does not need to crop the image...
+			if ( imgW <= dstW ) {
+				return false;
+			}
+
+			return true;
+		};
+
+		/* NOTE: Need to set this up every time instead of reusing if already there
+		as the toolbar button does not get reset when doing the following:
+		
+		simple_local_avatar_frame.setState('library');
+		simple_local_avatar_frame.open();
+		*/
+
+		simple_local_avatar_frame = wp.media({
+			button: {
+				text: i10n_SimpleLocalAvatars.selectCrop, // l10n.selectAndCrop,
+				close: false,
+			},
+			states: [
+				new wp.media.controller.Library({
+					title: i10n_SimpleLocalAvatars.selectCrop, // l10n.selectAndCrop,
+					library: wp.media.query({ type: 'image' }),
+					multiple: false, // We set multiple to false so only get one image from the uploader
+					date: false,
+					priority: 20,
+					suggestedWidth: 200,
+					suggestedHeight: 200,
+				}),
+				new wp.media.controller.CustomizeImageCropper({
+					imgSelectOptions: myTheme_calculateImageSelectOptions,
```

Comment:
I think here and below on the definition, we shouldn't namespace the function with `myTheme`